Cython/Compiler/ModuleNode.py

```diff
@@ -2928,14 +2929,11 @@ def generate_module_state_clear(self, env, code):
             Naming.empty_bytes)
         code.putln('Py_CLEAR(clear_module_state->%s);' %
             Naming.empty_unicode)
-        code.putln('#ifdef __Pyx_CyFunction_USED')
-        code.putln('Py_CLEAR(clear_module_state->%s);' %
-            Naming.cyfunction_type_cname)
-        code.putln('#endif')
-        code.putln('#ifdef __Pyx_FusedFunction_USED')
-        code.putln('Py_CLEAR(clear_module_state->%s);' %
-            Naming.fusedfunction_type_cname)
-        code.putln('#endif')
+        for tp, linked_types in Naming.type_names_and_linked_types:
```

Comment:
I think this clean-up is worth doing since it makes sure we handle all the shared types in clear and traverse (and not just CyFunction and FusedFunction). But in principle I could separate it out.

Reply:
Can't we move these to new utility code sections? That would put them close to the places where the globals are created, instead of letting multiple places know the complete list of such globals plus their (potentially changing) dependencies. It would also avoid all this `#ifdef …_USED`.

Reply:
Yes that might make sense. Essentially what I originally did with AsyncGen and undid with this PR....

I think it's just "traverse" that doesn't currently have a utility code section, but it's easy enough to add one

Reply:
I've done that. I think the old "cleanup" mechanism and module state clear needs rationalizing at some point, but I don't think this PR makes it any worse.